tipi/vault.hpp

```diff
@@ -33,6 +35,22 @@ BOOST_FUSION_ADAPT_STRUCT(tipi::auth_t, user, pass, endpoint, type);
 
 namespace tipi {
 
+  inline std::string decode64(const std::string &val) {
+    using namespace boost::archive::iterators;
+    using It = transform_width<binary_from_base64<std::string::const_iterator>, 8, 6>;
+    // See https://svn.boost.org/trac10/ticket/5629#comment:9
+    // Boost binary_from_base64 transforms '=' into '\0', they need to be removed to support binary data
+    auto padding_count = std::count(val.end() - std::min(std::size_t{2}, val.size()), val.end() , '=');
+    return std::string(It(std::begin(val)), It(std::end(val) - padding_count));
+  }
+
+  inline std::string encode64(const std::string &val) {
+    using namespace boost::archive::iterators;
+    using It = base64_from_binary<transform_width<std::string::const_iterator, 6, 8>>;
+    auto tmp = std::string(It(std::begin(val)), It(std::end(val)));
+    return tmp.append((3 - val.size() % 3) % 3, '=');
+  }
+
```

Comment:
Why would we not use xxhr ?

Reply:
because boringssl is really complicated to support with wasm-asmjs as target

Reply:
xxhr doesnt' need boringssl in emscripten builds, it might be wrongfully stated with the current deps file but there is a custom backend of xxhr for asmjs

Reply:
I will change that nd try to keep xxhr

Reply:
But I have in issue in how I can write the deps file. I think I can't says that deps is only for linux and this one for wasm. Or can we do that in tipi ?

Reply:
Done